src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -165,15 +165,7 @@ public IEnumerable<DiagnosticData> ConvertToLocalDiagnostics(Document targetDocu
                 Logger.Log(FunctionId.Diagnostics_ProjectDiagnostic, p => $"Failed to Load Successfully ({p.FilePath ?? p.Name})", project);
 
                 // get rid of any result except syntax from compiler analyzer result
-                var newCompilerAnalysisResult = new DiagnosticAnalysisResult(
-                    analysisResult.ProjectId,
-                    analysisResult.Version,
-                    analysisResult.SyntaxLocals,
-                    semanticLocals: ImmutableDictionary<DocumentId, ImmutableArray<DiagnosticData>>.Empty,
-                    nonLocals: ImmutableDictionary<DocumentId, ImmutableArray<DiagnosticData>>.Empty,
-                    others: ImmutableArray<DiagnosticData>.Empty,
-                    documentIds: null,
-                    fromBuild: false);
+                var newCompilerAnalysisResult = analysisResult.DropExceptSyntax();
```

Comment:
Cute.   I'm also feelign taht a .WithNothingButSyntax might be a better name.  The reason for this (to me at least) is that it follows the 'With' namign pattern that clearly indicates that you need to use the result.  'DropXXX' sounds very imperative and mutable, and makes me think that actual instance is mutating itself.  

Thoughts?

Reply:
If you do like 'Drop', i would be ok with WithAllDroppedExceptSyntax.

Reply:
Alternatively.  could this be `analysisResult.Reset().WithSyntax(...)`?